client_integration_test.go

```diff
@@ -651,6 +651,50 @@ func TestClientRemove(t *testing.T) {
 	}
 }
 
+func TestRemoveAll(t *testing.T) {
+	sftp, cmd := testClient(t, READWRITE, NODELAY)
+	defer cmd.Wait()
+	defer sftp.Close()
+
+	// Create a temporary directory for testing
+	tempDir, err := ioutil.TempDir("", "sftptest-removeAll")
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer os.RemoveAll(tempDir)
+
+	// Create file and directory within the temporary directory
+	f, err := ioutil.TempFile(tempDir, "sftptest-removeAll*.txt")
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer f.Close()
```

Comment:
If you’re just creating the file, and not writing anything to it, then you do not need to `defer` this. I think we also have helper functions available to write out temporary files?